web/src/app.js

```diff
@@ -26,56 +34,168 @@ konnektApp.config(['$routeProvider', function ($routeProvider) {
       templateUrl: 'registration.html',
       controller: 'registrationController',
     })
+    .when('/dashboard', {
+      templateUrl: 'dashboard.html',
+      controller: 'dashboardController',
+    })
     .otherwise({
       redirectTo: '/login',
     });
 }]);
 
 
-konnektApp.controller('registrationController', ['$scope', '$http', function ($scope, $http) {
+// REGISTER LISTENER - watch for route changes, this event will fire every time the route changes
+konnektApp.run(['$rootScope', '$location', 'UserService', function ($rootScope, $location, UserService) {
+
+  $rootScope.$on('$routeChangeStart', function (event, next, prev) {
+
+    if (next.$$route) {
+      if (next.$$route.originalPath === '/register') {
+      $location.path('/register');
+      } else if (!UserService.isLoggedIn()) {
+      $location.path('/login');
+      }
+    }
+  });
+}]);
+
+// FACTORIES
+konnektApp.factory('HttpService', ['$http', function ($http) {
+
+  function login(userData) {
+    return $http.post(`${appUrl}/login`, JSON.stringify(userData));
+  }
+
+  function register(userData) {
+    return $http.post(`${appUrl}/register`, JSON.stringify(userData));
+  }
+
+  return {
+    login: login,
+    register: register,
+  };
+}]);
+
+
+konnektApp.factory('UserService', ['HttpService', '$window', function (HttpService, $window) {
+
+  var getuserdata = {
```

Comment:
name suggests it's something that gets user data, but it's just an object of user data, so `userData` would be more descriptive
+ add a `getUserData()` method that returns this and use that in your controllers instead of directly using `userData`
+ add a `setUserData(newUserData)` method that updates this and use that in your controllers instead of directly modifying `userData`